test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,890 @@
+
+            $exception.FullyQualifiedErrorId | Should Match "CannotDefineNewType"
+        }
+
+        It "Verifies Add-Type works back in full language mode again" {
+            Add-Type -TypeDefinition 'public class AfterFullLanguageTest { public static string Hello = "HelloAfter"; }'
+            [AfterFullLanguageTest]::Hello | Should Be "HelloAfter"
+        }
+
+        AfterEach {
+            Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+        }
+    }
+
+    Describe "New-Object in constrained language" -Tags 'Feature' {
+
+        Context "New-Object with dotNet types" {
+
+            It "Verifies New-Object works in constrained language of allowed string type" {
+
+                $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+                $resultString = New-Object System.String "Hello"
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguage
+
+                $resultString | Should Be "Hello"
+            }
+
+            It "Verifies New-Object throws error in constrained language for disallowed IntPtr type" {
+
+                $ExecutionContext.SessionState.LanguageMode = "ConstrainedLanguage"
+
+                try
+                {
+                    New-Object System.IntPtr 1234
+                    throw "No Exception!"
+                }
+                catch
+                {
+                    $exception = $_
+                }
+
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguage
+
+                $exception.FullyQualifiedErrorId | Should Match "CannotCreateTypeConstrainedLanguage"
+            }
+
+            It "Verifies New-Object works for IntPtr type back in full language mode again" {
+
+                New-Object System.IntPtr 1234 | Should Be 1234
+            }
+        }
+
+        Context "New-Object with COM types" {
+
+            # TODO
+            # Test is pending because PowerShell Core currently stubs out Windows system lockdown APIs
+            It "Verifies New-Object with COM types is disallowed in system lock down" -Pending {
+
+                Set-SystemLockDown
+
+                try
+                {
+                    New-Object -Com ADODB.Parameter
+                    throw "No Exception!"
+                }
+                catch
+                {
+                    $exception = $_
+                }
+
+                Remove-SystemLockDown
+
+                $exception.FullyQualifiedErrorId | Should Match "CannotCreateComTypeConstrainedLanguage"
+            }
+
+            It "Verifies New-Object with COM types works back in full language mode again" {
+
+                $result = New-Object -ComObject ADODB.Parameter
+                $result.Direction | Should Be 1
+            }
+        }
+
+        AfterEach {
+            Remove-SystemLockDown
+        }
+    }
+
+    Describe "Mkdir command on function drive in constrained language" -Tags 'Feature' {

```

Comment:
is the mkdir function the point of this, or should it be using `new-item -itemtype directory`?

Reply:
Mkdir.  This is a function test.

Reply:
Nope, you're right it should be new-item